### ISSUE TITLE
Failed to build gem native extension 0.2.3

### ISSUE THREAD

**Issue by marcamillion** [discourse/mini_racer#137]
Hi Guys,
    I am trying to install mini_racer 0.2.3 on a machine with the following software versions:

- Ubuntu 16.04.6 LTS (GNU/Linux 4.4.0-145-generic x86_64)
- Ruby 2.3.0
- Rails 3.2.22.5

Unfortunately I can't upgrade Rails to a higher version, I am stuck with that version. 

This is the error I get:
```
Fetching: libv8-6.7.288.46.1-x86_64-linux.gem (100%)
Successfully installed libv8-6.7.288.46.1-x86_64-linux
Fetching: mini_racer-0.2.3.gem (100%)
Building native extensions. This could take a while...
ERROR:  Error installing mini_racer:
	ERROR: Failed to build gem native extension.

    current directory: /usr/local/lib/ruby/gems/2.3.0/gems/mini_racer-0.2.3/ext/mini_racer_extension
/usr/local/bin/ruby -r ./siteconf20190419-23653-1pqux8q.rb extconf.rb
checking for main() in -lpthread... yes
creating Makefile

current directory: /usr/local/lib/ruby/gems/2.3.0/gems/mini_racer-0.2.3/ext/mini_racer_extension
make "DESTDIR=" clean

current directory: /usr/local/lib/ruby/gems/2.3.0/gems/mini_racer-0.2.3/ext/mini_racer_extension
make "DESTDIR="
compiling mini_racer_extension.cc
In file included from /usr/local/include/ruby-2.3.0/ruby.h:33:0,
                 from mini_racer_extension.cc:2:
mini_racer_extension.cc: In function ‘VALUE protected_callback(VALUE)’:
/usr/local/include/ruby-2.3.0/ruby/ruby.h:882:52: error: ‘__builtin_choose_expr’ was not declared in this scope
      OBJ_WB_UNPROTECT((VALUE)(obj)), ((VALUE)(obj))))
                                                    ^
/usr/local/include/ruby-2.3.0/ruby/ruby.h:1039:48: note: in definition of macro ‘RARRAY_CONST_PTR’
 #define RARRAY_CONST_PTR(a) rb_array_const_ptr(a)
                                                ^
/usr/local/include/ruby-2.3.0/ruby/ruby.h:1059:50: note: in expansion of macro ‘RB_OBJ_WB_UNPROTECT_FOR’
 #define RARRAY_PTR(a) ((VALUE *)RARRAY_CONST_PTR(RB_OBJ_WB_UNPROTECT_FOR(ARRAY, a)))
                                                  ^
mini_racer_extension.cc:857:9: note: in expansion of macro ‘RARRAY_PTR’
         RARRAY_PTR(data->ruby_args));
         ^
Makefile:206: recipe for target 'mini_racer_extension.o' failed
make: *** [mini_racer_extension.o] Error 1

make failed, exit code 2

Gem files will remain installed in /usr/local/lib/ruby/gems/2.3.0/gems/mini_racer-0.2.3 for inspection.
```

**Comment by ignisf** [comment]
Hi, do you have `ruby-dev` and `build-essentials` installed?

**Comment by ignisf** [comment]
It may be a bug in Ruby 2.3.0, check this out https://bugs.ruby-lang.org/issues/11962

**Comment by marcamillion** [comment]
@ignisf Interesting. How can I find out if it was fixed in future releases of 2.3.x and which release specifically? Hard to follow that issue on ruby-lang.

**Comment by ignisf** [comment]
@marcamillion, Ruby 2.3 is past its EOL and is no longer maintained. Please consider updating as we don't support Ruby versions past their EOL.

If you prefer to use a package, take a look at https://www.brightbox.com/docs/ruby/ubuntu/ which offers up to date versions of Ruby for Ubuntu.

> @ignisf Interesting. How can I find out if it was fixed in future releases of 2.3.x and which release specifically? Hard to follow that issue on ruby-lang.

To answer your question specifically, it seems to me it landed in the .1 release.

**Comment by marcamillion** [comment]
So I tried using `Ruby 2.3.7` and that worked.

@ignisf I know most people/projects no longer support Ruby 2.3....but unfortunately I need it for this particular project :(

Thanks though.

**Comment by SamSaffron** [comment]
@marcamillion the old gems should all work ... what you need to do in your gemfile is just pin back the libv8 and mini_racer gem to ones that were released prior to 2.3 end of life-ing.
